assets/javascripts/modules/analytics/awin.js

```diff
@@ -1,27 +1,54 @@
 //Implementation docs here: https://drive.google.com/drive/folders/19AOGPkPbbFjqPJTdCcCba3EwpLiSHBwW
 define([
     'modules/analytics/analyticsEnabled',
-    'modules/analytics/dntEnabled'
-], function (analyticsEnabled, dntEnabled) {
+    'modules/analytics/dntEnabled',
+    'utils/cookie'
+], function (analyticsEnabled, dntEnabled, cookie) {
     'use strict';
 
+    const referrerCookieName = 'gu_referrer_channel';
+
     function init() {
-        var AWIN = {};
+        storeChannel();
+
+        if (guardian.pageInfo.pageType !== 'Thankyou') {
+            return;
+        }
+
+        let AWIN = {};
         AWIN.Tracking = {};
         AWIN.Tracking.Sale = {};
 
         /*** Set your transaction parameters ***/
-        const productData = guardian.pageInfo.productData
+        const productData = guardian.pageInfo.productData;
         AWIN.Tracking.Sale.amount = productData.amount;
         AWIN.Tracking.Sale.orderRef = productData.subscriptionId;
         AWIN.Tracking.Sale.parts = productData.productSegment + ':' + productData.amount;
         AWIN.Tracking.Sale.voucher = productData.promoCode;
         AWIN.Tracking.Sale.currency = productData.currency;
         AWIN.Tracking.Sale.test = '0';
-        AWIN.Tracking.Sale.channel = 'aw';
+        AWIN.Tracking.Sale.channel = getAwinChannel();
         window.AWIN = AWIN;
     }
 
+    function getAwinChannel() {
+        const channel = cookie.getCookie(referrerCookieName);
+        const values = channel ? channel.split('&') : null;
+        if (values && values[0] === 'afl' && values[1] === 'awin'){
+            return 'aw';
+        }
+        return 'na';
+    }
+
+    function storeChannel() {
+        const parsedUrl = new URL(window.location.href);
+        const utmSource = parsedUrl.searchParams.get('utm_source');
+        const utmMedium = parsedUrl.searchParams.get('utm_medium');
```

Comment:
Does we have polyfills for [this](https://developer.mozilla.org/en-US/docs/Web/API/URL#Browser_compatibility) on IE, Edge, old Safari in this codebase?

Reply:
I tested on IE11 & Safari 6.2 and it works

Reply:
I guess the answer is yes then, 👍.